backend/project/command-query/src/main/kotlin/com/novatecgmbh/eventsourcing/axon/project/task/command/Task.kt

```diff
@@ -111,12 +113,19 @@ class Task : BaseAggregate() {
   fun handle(command: CompleteTaskCommand): TaskId {
     when (status) {
       PLANNED -> throw IllegalStateException("Task has not yet been started.")
-      STARTED -> apply(TaskCompletedEvent(aggregateIdentifier))
+      STARTED -> {
+        if (isAnyTodoNotDone()) {
```

Comment:
Instead of an IllegalStateException, we could throw a "PreconditionFailedException" or something similar. This way we could differentiate in the controller, which http return code is used. An IllegalStateException is typically a state, that should never be reached and therefore often leads to a 500 error. This should rather become a 412 error.